src/zenml/integrations/smtp_email/alerters/smtp_email_alerter.py

```diff
@@ -0,0 +1,398 @@
+        """
+        return cast(SMTPEmailAlerterConfig, self._config)
+
+    @property
+    def settings_class(self) -> Type[SMTPEmailAlerterSettings]:
+        """Settings class for the SMTP Email alerter.
+
+        Returns:
+            The settings class.
+        """
+        return SMTPEmailAlerterSettings
+
+    def _get_recipient_email(
+        self, params: Optional[BaseAlerterStepParameters] = None
+    ) -> str:
+        """Get the recipient email address to use.
+
+        Args:
+            params: Optional parameters.
+
+        Returns:
+            Email address of the recipient.
+
+        Raises:
+            RuntimeError: if config is not of type `BaseAlerterStepParameters`.
+            ValueError: if an email recipient was neither defined in the config
+                nor in the alerter component.
+        """
+        if params and not isinstance(params, BaseAlerterStepParameters):
+            raise RuntimeError(
+                "The config object must be of type `BaseAlerterStepParameters`."
+            )
+        if (
+            params
+            and isinstance(params, SMTPEmailAlerterParameters)
+            and hasattr(params, "recipient_email")
+            and params.recipient_email is not None
+        ):
+            return params.recipient_email
+
+        try:
+            settings = cast(
+                SMTPEmailAlerterSettings,
+                self.get_settings(get_step_context().step_run),
+            )
+        except RuntimeError:
+            settings = None
+
+        if settings is not None and settings.recipient_email is not None:
+            return settings.recipient_email
+
+        if self.config.recipient_email is not None:
+            return self.config.recipient_email
+
+        raise ValueError(
+            "The `recipient_email` is not set either in the runtime settings, "
+            "or the component configuration of the alerter. Please specify at "
+            "least one."
+        )
+
+    def _should_include_html(
+        self, params: Optional[BaseAlerterStepParameters] = None
+    ) -> bool:
+        """Check if HTML content should be included.
+
+        Args:
+            params: Optional parameters.
+
+        Returns:
+            Boolean indicating if HTML should be included.
+        """
+        if (
+            params
+            and isinstance(params, SMTPEmailAlerterParameters)
+            and hasattr(params, "include_html")
+            and params.include_html is not None
+        ):
+            return params.include_html
+
+        try:
+            settings = cast(
+                SMTPEmailAlerterSettings,
+                self.get_settings(get_step_context().step_run),
+            )
+        except RuntimeError:
+            settings = None
+
+        if settings is not None and settings.include_html is not None:
+            return settings.include_html
+
+        return self.config.include_html
+
+    def _get_subject_prefix(self) -> str:
```

Comment:
All of these helper methods seem to be doing the same exact thing, can we merge that into a single method that takes the attribute name (and potentially a `fail_if_missing`) as arguments?